File: dsa_algo/leetcode/arrays/arrays_hash/14-longest_common_prefix.cpp
std::string longestCommonPrefix(vector<string>& strs) {
    std::string result = "";
    for(int i = 0; i < strs[0].size(); i++)
    {
        for(int s = 0; s < strs.size(); s++)
        {
            if( i == strs[s].size() || strs[s][i] != strs[0][i])
                return result;
        }
        result += strs[0][i];
    }
    return result;
}

File: DSAlgorithms/Leetcode/ts/src/arrays_hash/14-longest_common_prefix.ts
const longestCommonPrefix = (strs: string[]): string =>{
    let result: string = "";
    for(let i = 0; i < strs[0].length; i++)
    {
        for(let s = 0; s < strs.length; s++)
        {
            if(i === strs[s].length || strs[s][i] !== strs[0][i])
                return result;
        }
        result+= strs[0][i];
    }
    return result;
};
